parser: Fix parsing one-line enum declarations

MacDue opened the pull request.

jntrnr commented on tests/parser/one-line-enum-no-comma-ending.jakt:
```diff
@@ -0,0 +1 @@
+enum Fruit { Apple }
```
this is valid Jakt - are you thinking we should make it invalid?

MacDue replied:
It was an assumption based off the error that seemed to indicate that variants should end with a comma or newline.
I'll correct that.

MacDue replied:
(Currently enums like `enum Fruit { Apple }` don't parse correctly, so I was not 100% sure on the correct parse)

MacDue commented:
I've changed the PR to now allow enum declarations like `enum Fruit { Apple }`, and parse them correctly.  

I've also added a new sample that failed to compile before these changes.

jntrnr commented:
thanks for the fix!
